ext/digest/sha2/lib/sha2.rb

```diff
@@ -11,7 +11,12 @@
 #   $Id$
 
 require 'digest'
-require 'digest/sha2.so'
+
+if RUBY_ENGINE == 'jruby'
```

Comment:
These selections at runtime don't feel nice.

Reply:
I think this is fine.  `require 'digest/sha1.so'` is already platform-dependent and it's just a single line, so separating just a few lines into two small platform-specific files wouldn't pay off.

Reply:
we could probably generate the right loader based on the build platform and avoid checking the condition at runtime, but I don't think there will be any noticeable difference in performance.
maybe in a separate PR? I don't want to increase the complexity of this PR more than necessary.

Reply:
@knu, do you mean CRuby-specific by platform-dependent?